position_in_partition.hh

```diff
@@ -326,6 +330,13 @@ public:
         return {before_clustering_row_tag_t(), std::move(ck)};
     }
 
+    // If given position is a clustering row position, returns a position
+    // right before it. Otherwise, returns it unchanged.
+    // The position "pos" must be a clustering position.
+    static position_in_partition before_key(position_in_partition_view pos) {
+        return {before_clustering_row_tag_t(), pos};
+    }
```

Comment:
I think it's working correctly for non-full keys. Can you give an example?

Reply:
I was just surprised this is not symmetric to `after_key()` w.r.t. non-full keys.
I think this suffers from the same problem: `before_key()` for a prefix, will yield a keys that is before all prefixed keys, not right before the key value. I think it should be symmetrical.

Reply:
For a prefix key, say (a, null), a position which is before the prefix is also right before the key, because you cannot insert any key in between before_all_prefixed({a, null}) and equal_to({a, null}).

Reply:
Ah, right.